### PR TITLE
SeqPropertyCreator simplified

### LINKED ISSUE #321: SeqPropertyCreators not created for Seqs nested in Properties
Invoking both the following examples result in runtime ISE with `Property was created without provided SeqPropertyCreator in scope. Make sure not to explicitly pass creators on property creation.` error message.
```
val groups = SeqProperty[Seq[Int]](Seq[Seq[Int]](1 to 4))
groups.elemProperties.head.asSeq[Int]
```
```
Property[Seq[Int]](1 to 4).asSeq[Int]
```
And they used to work in Udash `0.7.1` - https://scalafiddle.io/sf/qpZvY74/0

### PR DESCRIPTION
**Pull request by ddworak**
Closes #321

### CODE REVIEW

File: core/src/main/scala/io/udash/properties/PropertyCreator.scala [hunk 1]
```diff
@@ -35,6 +38,12 @@ object PropertyCreator extends PropertyCreatorImplicits {
   implicit val Byte: PropertyCreator[Byte] = new SinglePropertyCreator[Byte]
   implicit val Boolean: PropertyCreator[Boolean] = new SinglePropertyCreator[Boolean]
   implicit val String: PropertyCreator[String] = new SinglePropertyCreator[String]
+
+  implicit final def materializeBSeq[T: PropertyCreator]: SeqPropertyCreator[T, BSeq] = new SeqPropertyCreator
```
[the remark this answers is not in the document]
Comment (ddworak) [reply core/src/main/scala/io/udash/properties/PropertyCreator.scala]: Just a flag in the bytecode in this case, but optimizing wherever I can. Will do.

File: core/src/main/scala/io/udash/properties/Properties.scala [hunk 2]
```diff
@@ -49,7 +49,7 @@ object Properties extends Properties {
   }
 
   class Any2SeqProperty[A] private[properties](private val value: Seq[A]) extends AnyVal {
-    def toSeqProperty[B >: A : PropertyCreator]: ReadableSeqProperty[B] = new ImmutableSeqProperty[B](value)
+    def toSeqProperty[B >: A : PropertyCreator]: ReadableSeqProperty[B] = new ImmutableSeqProperty[B, Seq](value)
```
[the remark this answers is not in the document]
Comment (ddworak) [reply core/src/main/scala/io/udash/properties/Properties.scala]: We could keep it, but we cannot convey it in the return signature anyway. Fixing the creator usage here anyway now that you've got my attention :D

File: core/src/main/scala/io/udash/properties/model/ReadableModelProperty.scala [hunk 3]
```diff
@@ -26,9 +26,8 @@ trait ReadableModelProperty[A] extends ReadableProperty[A] {
 }
 
 trait ModelPropertyMacroApi[A] extends ReadableModelProperty[A] {
-  def getSubProperty[T](getter: A => T, key: String): ReadableProperty[T]
+  def getSubProperty[T: PropertyCreator](getter: A => T, key: String): ReadableProperty[T]
   def getSubModel[T](getter: A => T, key: String): ReadableModelProperty[T]
```
Comment (bgrochal) [inline core/src/main/scala/io/udash/properties/model/ReadableModelProperty.scala]: Should `T` be equipped with `ModelPropertyCreator`?

File: core/src/main/scala/io/udash/properties/PropertyCreator.scala [hunk 4]
```diff
@@ -27,36 +30,46 @@ object PropertyCreator extends PropertyCreatorImplicits {
...
+  override def newImmutableProperty(value: T): ImmutableProperty[T] = new ImmutableProperty[T](value)
 }
 
-object SeqPropertyCreator {
-  implicit def materializeSeq[T: PropertyCreator, SeqTpe[T] <: Seq[T]](
-    implicit cbf: CanBuildFrom[Nothing, T, SeqTpe[T]]
-  ): SeqPropertyCreator[T, SeqTpe] = macro io.udash.macros.PropertyMacros.reifySeqPropertyCreator[T, SeqTpe]
+final class SeqPropertyCreator[A: PropertyCreator, SeqTpe[T] <: Seq[T]](implicit cbf: CanBuildFrom[Nothing, A, SeqTpe[A]])
+  extends PropertyCreator[SeqTpe[A]] {
+  protected def create(prt: ReadableProperty[_]): CastableProperty[SeqTpe[A]] =
```
Comment (FreexD) [inline core/src/main/scala/io/udash/properties/PropertyCreator.scala]: what does `prt` stand for?

File: core/src/main/scala/io/udash/properties/PropertyCreator.scala [hunk 5]
```diff
@@ -27,36 +30,46 @@ object PropertyCreator extends PropertyCreatorImplicits {
...
-  implicit val Boolean: PropertyCreator[Boolean] = new SinglePropertyCreator[Boolean]
-  implicit val String: PropertyCreator[String] = new SinglePropertyCreator[String]
+  implicit final val Double: PropertyCreator[Double] = new SinglePropertyCreator
+  implicit final val Float: PropertyCreator[Float] = new SinglePropertyCreator
+  implicit final val Long: PropertyCreator[Long] = new SinglePropertyCreator
+  implicit final val Int: PropertyCreator[Int] = new SinglePropertyCreator
+  implicit final val Short: PropertyCreator[Short] = new SinglePropertyCreator
+  implicit final val Byte: PropertyCreator[Byte] = new SinglePropertyCreator
+  implicit final val Boolean: PropertyCreator[Boolean] = new SinglePropertyCreator
+  implicit final val String: PropertyCreator[String] = new SinglePropertyCreator
```
Comment (FreexD) [inline core/src/main/scala/io/udash/properties/PropertyCreator.scala]: no `PropertyCreator` for `Char`?

File: macros/src/main/scala/io/udash/macros/PropertyMacros.scala [hunk 6]
```diff
@@ -301,10 +275,14 @@ class PropertyMacros(val ctx: blackbox.Context) extends AbstractMacroCommons(ctx
   }
 
   private def subSeqValidation(leafTpe: Type): Unit = {
-    if (!isSeqPropertyTpe(leafTpe)) c.abort(c.enclosingPosition, s"$leafTpe is not a valid subSeq type")
+    val singleParamSeq: Boolean = {
+      val dealiased = leafTpe.map(_.dealias)
+      dealiased.typeConstructor <:< SeqTpe.typeConstructor && dealiased.typeConstructor.typeParams.lengthCompare(1) == 0
```
Comment (FreexD) [inline macros/src/main/scala/io/udash/macros/PropertyMacros.scala]: no need for `dealiased.typeParams.lengthCompare(1) == 0` check as well?

Comment (ddworak) [reply macros/src/main/scala/io/udash/macros/PropertyMacros.scala]: Nope, type constructor has type params, type has type args. By explicitly getting the type constructor I know what to check.

**Pull request merged**